src/router.js

```diff
@@ -138,56 +141,48 @@ function createLoaderForSchema(schema, actionCreatorForLoadingObject, resetActiv
         // Fire load action for the event program program to be edited
         store.dispatch(actionCreatorForLoadingObject({ schema, id: params.modelId }));
         store.dispatch(resetActiveStep());
-
-        callback();
     };
 }
 
-function loadList({ params }, replace, callback) {
+function loadList({ params }, replace) {
     if (params.modelType === 'organisationUnit') {
-        // Don't load organisation units as they get loaded through the appState
-        // Also load the initialState without cache so we refresh the assigned organisation units
-        // These could have changed by adding an organisation unit which would need to be reflected in the
-        // organisation unit tree
+        // Don't load organisation units as they get loaded through the
+        // appState Also load the initialState without cache so we
+        // refresh the assigned organisation units These could have
+        // changed by adding an organisation unit which would need to be
+        // reflected in the organisation unit tree
         initState({ params });
-        return callback();
     }
 
     initState({ params });
     return listActions.loadList(params.modelType)
         .take(1)
         .subscribe(
```

Comment:
I'm pretty new to RxJS so I'm not sure if it makes sense to subscribe here anymore to capture the error messages, but I added a `noop` function to allow our error handling to function as it did before.

Reply:
I believe it does, as the loadList, and similar functions are still async (but handled by observables) and can throw an error, so this makes sense to me!